src/data/puzzles/Adventure.tsx

```diff
@@ -0,0 +1,397 @@
+                </summary>
+            </RulesParagraph>
+            
+            <RulesParagraph>
+                <RulesUnorderedList>
+                    {rules.map(item => <li>{item}</li>)}
+                </RulesUnorderedList>
+            </RulesParagraph>
+        </>
+    );
+}
+
+export const ChooseYourOwnAdventure: PuzzleDefinition<AdventurePTM<number>> = {
+    
+    title: { [LanguageCode.en]: "Adventure is out there!" },
+    author: { [LanguageCode.en]: "Tumbo" },
+    extension: {
+        rootChoiceTaken: adventureDef
+    },
+    slug: "choose-your-own-adventure",
+    initialDigits: { 6: { 0: 1 }, 7: { 4: 1}, 8: { 7: 2, 8: 9}},
+    rules: getAdventureRules,
+    typeManager: AdventureTypeManager(),
+    gridSize: GridSize9,
+    regions: Regions9,
+    solution: createCellsMapFromArray([
+        [9, 1, 4, 2, 8, 7, 6, 5, 3],
+        [3, 7, 5, 1, 9, 6, 4, 8, 2],
+        [8, 2, 6, 3, 5, 4, 9, 1, 7],
+        [2, 4, 9, 7, 3, 1, 8, 6, 5],
+        [5, 6, 8, 9, 4, 2, 3, 7, 1],
+        [7, 3, 1, 8, 6, 5, 2, 9, 4],
+        [1, 8, 7, 4, 2, 9, 5, 3, 6],
+        [6, 9, 2, 5, 1, 3, 7, 4, 8],
+        [4, 5, 3, 6, 7, 8, 1, 2, 9],
+    ]),
+    resultChecker: (context) => {
+        return isValidFinishedPuzzleByEmbeddedSolution(context);
+    },
+    successMessage: (context) => `You are exhausted having fully filled in your map. The sun is getting low and you'll need to hurry to make it home before curfew. You can't wait to tell your parents about the ${context.gridExtension.choicesMadeSolutionStrings[0]}, ${context.gridExtension.choicesMadeSolutionStrings[1]}, and ${context.gridExtension.choicesMadeSolutionStrings[2]}!`,
+    items: (context) => {
+        return [
+            ...getAdventureConstraints(context),
+        ].map(toDecorativeConstraint);
+    },
+    /* lmdLink: "TODO",
+    getLmdSolutionCode: ({ puzzle: { solution } }) =>
+        indexes(9)
+            .map((index) => solution![0][index])
+            .join(""),*/
+}
+
+const getAdventureConstraints = (context: PuzzleContext<AdventurePTM>): Constraint<AdventurePTM, any>[] => {
+    let constraints: Constraint<AdventurePTM, any>[] = [];
+    let currentChoice: choiceTaken | undefined = context.puzzle.extension.rootChoiceTaken;
+    var depth = 0;
+    while (currentChoice !== undefined)
+    {
+        constraints = constraints.concat(currentChoice.constraints);
+        if (currentChoice.choices !== undefined && (context.gridExtension.choicesMade.length === depth || context.gridExtension.choicesMade.length === depth + 1))
+        {
+            var solved = checkSolved(context, currentChoice.choices.solveCells)
+            if (context.gridExtension.choicesMade.length === depth + 1 && !solved)
+            {
+                context.gridExtension.choicesMade.pop();
+                context.gridExtension.choicesMadeSolutionStrings.pop();
```

Comment:
The `pop()` method modifies the array in-place. The contents of `context` shouldn't be modified.